dbms/src/DataStreams/CreatingSetsBlockInputStream.cpp

```diff
@@ -291,4 +291,16 @@ void CreatingSetsBlockInputStream::createOne(SubqueryForSet & subquery)
     }
 }
 
+uint64_t CreatingSetsBlockInputStream::collectCPUTimeNsImpl(bool /*is_thread_runner*/)
+{
+    // CreatingSetsBlockInputStream isn't a thread-runner, so is_thread_runner is ignored here.
+    // But each of CreatingSetsBlockInputStream's children is a thread-runner.
+    uint64_t cpu_time_ns = 0;
+    forEachChild([&](IBlockInputStream & child) {
+        cpu_time_ns += child.collectCPUTimeNs(true);
```

Comment:
Actually, the last child is not thread runner.

Reply:
I think it depends on what stream the last child is, if it is a computational stream, it will also consume a lot of cpu on the `mpptaskimpl` thread.